ieee-cdf/ieee-cdf-converter/src/main/java/com/powsybl/ieeecdf/converter/IeeeCdfImporter.java

```diff
@@ -0,0 +1,551 @@
+                            .setTargetQ(ieeeCdfBus.getReactiveGeneration())
+                            .setVoltageRegulatorOn(false)
+                            .add();
+                    break;
+
+                case HOLD_VOLTAGE_WITHIN_VAR_LIMITS:
+                case HOLD_VOLTAGE_AND_ANGLE:
+                    Generator generator = newGeneratorAdder(ieeeCdfBus, voltageLevel)
+                            .setTargetV(ieeeCdfBus.getDesiredVoltage() * voltageLevel.getNominalV())
+                            .setVoltageRegulatorOn(true)
+                            .add();
+                    if (ieeeCdfBus.getMinReactivePowerOrVoltageLimit() != 0 || ieeeCdfBus.getMaxReactivePowerOrVoltageLimit() != 0) {
+                        generator.newMinMaxReactiveLimits()
+                                .setMinQ(ieeeCdfBus.getMinReactivePowerOrVoltageLimit())
+                                .setMaxQ(ieeeCdfBus.getMaxReactivePowerOrVoltageLimit())
+                                .add();
+                    }
+                    break;
+
+                default:
+                    throw new IllegalStateException("Unexpected bus type: " + ieeeCdfBus.getType());
+            }
+        }
+    }
+
+    private static Bus createBus(IeeeCdfBus ieeeCdfBus, VoltageLevel voltageLevel) {
+        String busId = getBusId(ieeeCdfBus.getNumber());
+        Bus bus = voltageLevel.getBusBreakerView().newBus()
+                .setId(busId)
+                .setName(ieeeCdfBus.getName())
+                .add();
+        bus.setV(ieeeCdfBus.getFinalVoltage() * voltageLevel.getNominalV())
+                .setAngle(ieeeCdfBus.getFinalAngle());
+        return bus;
+    }
+
+    private static Substation createSubstation(Network network, String substationId) {
+        Substation substation = network.getSubstation(substationId);
+        if (substation == null) {
+            substation = network.newSubstation()
+                    .setId(substationId)
+                    .add();
+        }
+        return substation;
+    }
+
+    private static double parseNominalVoltageInBusName(IeeeCdfBus ieeeCdfBus) {
+        if (ieeeCdfBus.getName().length() < 9) {
+            throw new PowsyblException("Bus name is too short for parsing nominal voltage");
+        }
+        String nominalVoltageStr = ieeeCdfBus.getName().substring(9, 12).trim();
+        switch (nominalVoltageStr) {
+            case "V1":
+            case "HV":
+                return 345;
+            case "V2":
+            case "LV":
+                return 138;
+            case "V3":
+                return 161;
+            case "V4":
+            case "TV":
+                return 33;
+            case "V5":
+                return 14;
+            case "V6":
+                return 11;
+            case "V7":
+            case "ZV":
+                return 1;
+            default:
+                throw new IllegalStateException("Unknown nominal voltage: '" + nominalVoltageStr + "'");
```

Comment:
Is this exception the good one? A PowsyblException is thrown if the name is invalid, why don’t throwing the same exception type here too?

Reply:
This method has been removed